src/linear_algebra/quadratic_form.lean

```diff
@@ -788,3 +789,229 @@ theorem exists_orthogonal_basis [hK : invertible (2 : K)]
 let ⟨v, hv₁, hv₂, _⟩ := exists_orthogonal_basis' hB₁ hB₂ in ⟨v, hv₁, hv₂⟩
 
 end bilin_form
+
+/-- Nonzero elements in a field are invertible. This is a definition since it requires a
+parameter. -/
+def field.invertible {K : Type*} [field K] {z : K} (hz : z ≠ 0) : invertible z :=
+{ inv_of := z⁻¹,
+  inv_of_mul_self := inv_mul_cancel hz,
+  mul_inv_of_self := mul_inv_cancel hz }
+
+namespace quadratic_form
+
+open_locale big_operators
+
+open finset bilin_form
+
+variables {M₁ : Type*} [add_comm_group M₁] [module R M₁]
+variables {ι : Type*} [fintype ι] {v : ι → M}
+
+/-- A quadratic form composed with a `linear_equiv` is isometric to itself. -/
+def isometry_of_comp_linear_equiv (Q : quadratic_form R M) (f : M₁ ≃ₗ[R] M) :
+  Q.isometry (Q.comp (f : M₁ →ₗ[R] M)) :=
+{ map_app' :=
+  begin
```

Comment:
is it too slow to just do `:= λ _, by simp`?

Reply:
I'm not sure, I thought we should use `simp only` to speed things up though it is not too slow here.
Also, it seems that its displaying code from #7416. Would you mind leaving your reviews there? I don't know how to make it so that code from there is not shown here :/